fix up meta for proper 2.7 builds

bollwyvl opened the pull request:
Fixes #2, hopefully!
Fixes #5

Not sure how this was even working before! This adds:
- `python` to all the build/run/test steps
- `setuptools` on build
- some command tests for `--version` and `--help`

Yeesh!

Commit:
fix up meta for proper 2.7 builds

Commit:
add version command test

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (`recipe`) and found it was in an excellent condition.

Commit:
add git to test deps

willingc commented:
Nicely done @bollwyvl

jakirkham commented on recipe/meta.yaml:
```diff
@@ -13,27 +13,34 @@ source:
 build:
-  number: 0
+  number: 1
   script: python setup.py install --single-version-externally-managed --record=record.txt
```
We might want to add [`entry_points`](http://conda.pydata.org/docs/building/meta-yaml.html#python-entry-points).

bollwyvl replied:
added!

jakirkham replied:
Thanks.

jakirkham committed:
Re-render with `conda-smithy` version `1.0.3`.

jakirkham commented:
Might also be a good idea to re-render too. Have opened PR ( https://github.com/conda-forge/cookiecutter-feedstock/pull/4 ), which re-render's with the latest `conda-smithy`.

jakirkham committed:
Merge remote-tracking branch 'bollwyvl/fix/help-27' into rerender_1.0.3

jakirkham committed:
recipe: Add `pip` as a test requirement.

jakirkham committed:
recipe: Fix-up test dependencies and simplify the recipe.

Commit:
remove pip deps

Commit:
used actual entrypoint called out in upstream setup.py

bollwyvl commented:
Added the entry_point...

Rerendered with smithy `1.0.3` on this PR so we don't have to increment the build a bunch of times...

Also removed the pip dependencies called out in #5, again for the same reason...

Here's hoping  🎰 !

jakirkham committed:
Merge remote-tracking branch 'bollwyvl/fix/help-27' into rerender_1.0.3

jakirkham commented:
Added a PR ( https://github.com/bollwyvl/cookiecutter-feedstock/pull/1 ) to your PR. Fixes up some other things. Some caused by doing the re-rendering. Others were related to the current testing strategy.

bollwyvl committed:
Merge pull request #1 from jakirkham/rerender_1.0.3

Fixes up some stuff

Commit:
bah, that was the entry_point from master. left a note for future self

Commit:
Merge remote-tracking branch 'origin/fix/help-27' into fix/help-27

jakirkham commented on recipe/meta.yaml:
```diff
@@ -13,27 +13,42 @@ source:
+  entry_points:
+    - cookiecutter = cookiecutter.cli:main
+    # master has changed it's setup a bit as of 4a64e48... will have to fix
+    # - cookiecutter = cookiecutter.__main__:main
```
Ah, ok, that makes sense. Was wondering why I was having issues with it.

bollwyvl commented:
@jakirkham thanks! Merged your simplification. That's great.

Also fixed the entrypoint to be the one associated with this release (not master) 😊

bollwyvl commented on recipe/meta.yaml:
```diff
@@ -13,27 +13,42 @@ source:
+    - cookiecutter --help
+    # Run test suite.
+    - cd {{ environ["SRC_DIR"] }}
+    - py.test --cov=cookiecutter -k "not _hg_ and not mercurial and not test_make_sure_path_exists and not pypackage_git"
```
I guess we could actually test with mercurial on py27... not right now, though :]

jakirkham replied:
Yeah, we could do some selector business. Seems fine to save it for a new issue and separate PR. This one is pretty busy anyways.

jakirkham commented:
Awesome! Thanks for your work on this.

Also, I'd be willing to help maintain if you wish to add me. Think I might be using this quite a bit in the near future.

bollwyvl commented:
@jakirkham of course, the more the merryer! I'll add you once the tests succeed/fail... hate giving up my spot in line at the appveyor ride 🎢 !

We're starting to fire up some more cookiecutter work over on Jupyter as well... granted, I do like the features afforded by the whole yeoman toolchain (mmm.... composability) but you can't beat cookiecutter for simplicity!

jakirkham commented:
> @jakirkham of course, the more the merryer! I'll add you once the tests succeed/fail... hate giving up my spot in line at the appveyor ride 🎢 !

No worries. I can make a different PR for that.

Edit: Please see PR ( https://github.com/conda-forge/cookiecutter-feedstock/pull/6 ).

bollwyvl commented:
@jakirkham Oh, it's probably worth adding you on this PR... tests have almost passed! hooray!

jakirkham commented:
Ok, whichever is easiest. 😄

jakirkham commented:
Passing. 😄 Not sure if you wanted to do more to it. Can go ahead and merge now if you want.

Commit:
add @jakirkham as maintainer

bollwyvl commented:
Hooray test win!

I guess we'll pay the test tax one more time, so we can put this flurry of activity to bed 😓 !

bollwyvl commented:
Hooray!

jakirkham commented:
Marked the old builds as broken as it appears they didn't list the Python version. 😱 Fortunately, the new ones do thanks to these changes.

bollwyvl commented:
Thanks!
